irqbalance.c

```diff
@@ -465,17 +478,21 @@ gboolean sock_handle(gint fd, GIOCondition condition, gpointer user_data __attri
 		if (!strncmp(buff, "setup", strlen("setup"))) {
 			char banned[512];
 			char *setup = calloc(strlen("SLEEP  ") + 11 + 1, 1);
+			char *old = NULL;
 
 			if (!setup)
 				goto out_close;
 			snprintf(setup, strlen("SLEEP  ") + 11 + 1, "SLEEP %d ", sleep_interval);
 			if(g_list_length(cl_banned_irqs) > 0) {
-				for_each_irq(cl_banned_irqs, get_irq_data, setup);
+				for_each_irq(cl_banned_irqs, get_irq_data, &setup);
 			}
 			cpumask_scnprintf(banned, 512, banned_cpus);
-			setup = realloc(setup, strlen(setup) + strlen(banned) + 7 + 1);
-			if (!setup)
+			old = setup;
+			setup = realloc(old, strlen(old) + strlen(banned) + 7 + 1);
+			if (!setup) {
+				free(old);
```

Comment:
Instead of freeing here, assign the temp variable to the result of realloc, and, if its _not_ null, assign it to *setup, then call sprintf, and send.  That way you only have a single call to free at the bottom of the function

Reply:
ok, I have modify the patch. thanks.